packages/payment-processor/test/payment/erc20-batch-proxy.test.ts

```diff
@@ -0,0 +1,508 @@
+        }
+
+        const request = Utils.deepCopy(validRequest);
+        request.extensions['pn-erc20-fee-proxy-contract'].values.feeAmount = '6';
+
+        // get the balance to compare after payment
+        const balanceEthBefore = await wallet.getBalance();
+        const balanceErc20Before = await getErc20Balance(validRequest, wallet.address, provider);
+        const feeBalanceErc20Before = await getErc20Balance(validRequest, feeAddress, provider);
+
+        const tx = await payBatchProxyRequest(
+          [request, validRequest],
+          batchVersion,
+          wallet,
+          batchFee,
+        );
+        const confirmedTx = await tx.wait(1);
+
+        const balanceEthAfter = await wallet.getBalance();
+        const balanceErc20After = await getErc20Balance(validRequest, wallet.address, provider);
+        const feeBalanceErc20After = await getErc20Balance(validRequest, feeAddress, provider);
+
+        expect(confirmedTx.status).toBe(1);
+        expect(tx.hash).not.toBeUndefined();
+        expect(balanceEthAfter.lte(balanceEthBefore)).toBeTruthy(); // 'ETH balance should be lower'
+
+        // ERC20 balance should be lower
+        expect(
+          BigNumber.from(balanceErc20After).eq(
+            BigNumber.from(balanceErc20Before).sub(2000 + (6 + 10) + (2 + 10)),
+          ),
+        ).toBeTruthy();
+        // fee ERC20 balance should be higher
+        expect(
+          BigNumber.from(feeBalanceErc20After).eq(
+            BigNumber.from(feeBalanceErc20Before).add(6 + 10 + (2 + 10)),
+          ),
+        ).toBeTruthy();
+      });
+    });
+
+    describe('prepareBatchPaymentTransaction', () => {
+      it('should consider the version mapping', () => {
+        expect(
+          prepareBatchPaymentTransaction(
+            [
+              {
+                ...validRequest,
+                extensions: {
+                  [PaymentTypes.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT]: {
+                    ...validRequest.extensions[
+                      PaymentTypes.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT
+                    ],
+                    version: '0.1.0',
+                  },
+                },
+              } as any,
+              {
+                ...validRequest,
+                extensions: {
+                  [PaymentTypes.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT]: {
+                    ...validRequest.extensions[
+                      PaymentTypes.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT
+                    ],
+                    version: '0.1.0',
+                  },
+                },
+              } as any,
+            ],
+            batchVersion,
+            batchFee,
+          ).to,
+        ).toBe(batchPaymentsArtifact.getAddress('private', '0.1.0'));
+      });
+    });
+  });
+
+  describe('encodePayBatchRequest: multi tokens -> using two ERC20', () => {
+    it('should throw an error if the request is not erc20, for two ERC20', async () => {
```

Comment:
There is a lot of repetition in your tests. Suggestion: you could make it shorter by doing something like bellow.
This is just a quick example but you could adapt this as you wish if you want to factorize your test suite.
```ts
const testSuite = (
  suiteName: string,
  requestTemplate1: ClientTypes.IRequestData,
  requestTemplate2: ClientTypes.IRequestData
) => {
  describe(suiteName, () => {
    let request1: ClientTypes.IRequestData;
    let request2: ClientTypes.IRequestData;
   
    beforeEach(() => {
      request1 = Utils.deepCopy(requestTemplate1) as ClientTypes.IRequestData;
      request2 = Utils.deepCopy(requestTemplate2) as ClientTypes.IRequestData;
    });

    it('test1', ...) // write your tests here directly using request1 and request2
    it('test2', ...)
  })
}

describe('erc20-batch-proxy', () => {
  testSuite('encodePayErc20BatchRequest for one type of ERC20', validRequest, validRequest);
  testSuite('encodePayErc20BatchRequest using two different ERC20', validRequest, fauValidRequest);
});
```